### DISCUSSION TITLE
Understand code training

### DISCUSSION THREAD

**Discussion by drnow4u** [codewars/codewars.com#2666] [Ideas]
As a developer, I spend most of the time reading and trying to understand my or others' code. I would like to have read code training.

I don't have the final proposition yet. My initial vision is:

You could imagine that randomly someone's source code solution for Kata is presented as a question. A student has to figure out what the presented code is doing. The answer could be multi-choice (Kata and solution are already correlated). For correct answer student will get scores and the author of the solution as well. At this moment for a user's solution, we can vote for 'Best Practices' and 'Clever'. We could have additionally `Readability` calculated automatically. In some cases, a clever solution could be difficult to understand.

Looking for your opinions and other proposition on the code readability.

**Comment by JohanWiltink** [comment]
Fun idea.

`Clever` solutions actually are almost never very readable - I like the idea of making readability a priority. Unreadable oneliners are overrated ( they have value, but not as much as they get now ).

**Comment by dylanson25** [comment]
I just start in coding wars and in the first Katas, I saw other codes and I had to search some functions or methods They used. However it is part of learn because I can understand how functions and methods work in toghether to generate a correct answer.
(I am learning english  so sorry if I bad wrote something )
